src/test/functional/functional-tests.js

```diff
@@ -733,8 +901,35 @@ describe('functional tests', function() {
       )
     })
 
-    it('should list objects', done => {
-      client.listObjects(bucketName, '', true)
+    step('listObjects(bucketName, prefix, recursive)_prefix: miniojsprefix, recursive:true_', done => {
+      client.listObjects(bucketName, listObjectPrefix, true)
+        .on('error', done)
+        .on('end', () => {
+          if (_.isEqual(objArray, listPrefixArray)) return done()
+          return done(new Error(`listObjects lists ${listPrefixArray.length} objects, expected ${listObjectsNum}`))
+        })
+        .on('data', data => {
+          listPrefixArray.push(data.name)
+        })
+    })
+
+    step('listObjects(bucketName, prefix, recursive)_bucketName: empty, recursive:true_', done => {
+      try {
+        client.listObjects("", "", true)
+          .on('end', () => {
+            return done(new Error(`listObjects should throw exception when empty bucketname is passed`))               
+          })
+      } catch (e) {
+          if (e.name == 'InvalidBucketNameError') {
```

Comment:
Perhaps it'd be simpler to use [`assert.throws(..., /InvalidBucketNameError/);`](http://chaijs.com/api/assert/#method_throws) here instead of the try / catch?

Reply:
@brendanashworth how do we call done() after doing assert.throws? we have to call `done()` or `done(new Error("listObjects should throw exception when empty bucketname is passed"))` depending on if listObjects throws the excpetion